packages/dds/sequence/src/intervalIndex/overlappingSequenceIntervalsIndex.ts

```diff
@@ -16,15 +16,16 @@ import {
 	SequenceInterval,
 	createPositionReferenceFromSegoff,
 } from "../intervals";
+import { SharedString } from "../sharedString";
 import { SequenceIntervalIndexes } from "./sequenceIntervalIndexes";
 import { OverlappingIntervalsIndex } from "./overlappingIntervalsIndex";
 
 class OverlappingSequenceIntervalsIndex
 	extends OverlappingIntervalsIndex<SequenceInterval>
 	implements SequenceIntervalIndexes.Overlapping
 {
-	constructor(client: Client) {
-		super(client, sequenceIntervalHelpers);
+	constructor(sharedString: SharedString) {
+		super(sharedString as unknown as Client, sequenceIntervalHelpers);
```

Comment:
this is not an appropriate case. shared string is not a client, it has a client property that is private which you need to access